### DISCUSSION TITLE
Disable Preview Deployments for all but two branches

### DISCUSSION THREAD

**Discussion by n-glaz** [vercel/vercel#8553] [Help] [answered]
Vercel Preview deployments are awesome, but our dev databases get really messy when WIP migrations get deployed by Vercel to one of our staging DBs and those migrations end up changing.

I only want to deploy certain branches (e.g. `master`, `prod`) on push. I've configured it so only those branches have environment variables corresponding to their databases, but Vercel still tries to deploy every single push and the deployment obv fails due to the lack of a DB credential.

It seems like a waste of resources to keep building every single push -- is there a way to disable Vercel preview deployments for all but a few branches via Vercel's GUI or CLI, or do I have to switch to manual deployment via hooks/actions in order to disable preview-on-push to any branch?

**Comment by amyegan** [comment, marked as answer]
You can use an [Ignored Build Step](https://vercel.com/docs/concepts/projects/overview#ignored-build-step) script to only allow deployment of specified branches. Here's a guide with a nice example of how you could do that: [How do I use the "Ignored Build Step" field on Vercel?](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel)
